x-pack/plugins/actions/server/lib/validate_with_schema.ts

```diff
@@ -24,33 +24,40 @@ type ValidKeys = 'params' | 'config' | 'secrets';
 function validateWithSchema(
   actionType: ActionType,
   key: ValidKeys,
-  value: any
-): Record<string, any> {
-  if (actionType.validate == null) return value;
-
-  let name;
-  try {
-    switch (key) {
-      case 'params':
-        name = 'action params';
-        if (actionType.validate.params == null) return value;
-        return actionType.validate.params.validate(value);
-
-      case 'config':
-        name = 'action type config';
-        if (actionType.validate.config == null) return value;
-        return actionType.validate.config.validate(value);
-
-      case 'secrets':
-        name = 'action type secrets';
-        if (actionType.validate.secrets == null) return value;
-        return actionType.validate.secrets.validate(value);
+  value: unknown
+): Record<string, unknown> {
+  if (actionType.validate) {
+    let name;
+    try {
+      switch (key) {
+        case 'params':
+          name = 'action params';
+          if (actionType.validate.params) {
+            return actionType.validate.params.validate(value);
+          }
+          break;
+        case 'config':
+          name = 'action type config';
+          if (actionType.validate.config) {
+            return actionType.validate.config.validate(value);
+          }
+
+          break;
+        case 'secrets':
+          name = 'action type secrets';
+          if (actionType.validate.secrets) {
+            return actionType.validate.secrets.validate(value);
+          }
+          break;
+        default:
+          // should never happen, but left here for future-proofing
+          throw new Error(`invalid actionType validate key: ${key}`);
+      }
+    } catch (err) {
+      // we can't really i18n this yet, since the err.message isn't i18n'd itself
+      throw Boom.badRequest(`error validating ${name}: ${err.message}`);
     }
-  } catch (err) {
-    // we can't really i18n this yet, since the err.message isn't i18n'd itself
-    throw Boom.badRequest(`error validating ${name}: ${err.message}`);
   }
 
-  // should never happen, but left here for future-proofing
-  throw new Error(`invalid actionType validate key: ${key}`);
+  return value as Record<string, unknown>;
```

Comment:
This function was refactored to avoid having to do the ` as Record<string, unknown>` casting 4 times in a single function, the logic is identical.